main.js

```diff
@@ -23,8 +65,19 @@
             const records = data.result.records;
             records.forEach((record, i) => {
                 console.log(i);
-                L.marker([record.Y, record.X]).addTo(map)
+
+                //Collect time of incident from the record
+                var incidentYear = parseInt(record.INCIDENTTIME.substring(0,4));
```

Comment:
So this would become `record.incidentYear = parseInt(record.INCIDENTTIME.substring(0,4));`, similar thing for the other two